apis/metal3.io/v1alpha1/bmceventsubscription_types.go

```diff
@@ -0,0 +1,92 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+
+	// BMCEventSubscriptionFinalizer is the name of the finalizer added to
+	// subscriptions to block delete operations until the subscription is removed
+	// from the BMC.
+	BMCEventSubscriptionFinalizer string = "bmceventsubscription.metal3.io"
+)
+
+type BMCEventSubscriptionSpec struct {
+	Taints []corev1.Taint `json:"taints,omitempty"`
+
+	// A reference to a BareMetalHost
+	HostRef string `json:"hostRef,omitempty"`
+
+	// A webhook URL to send events to
+	Destination string `json:"destination,omitempty"`
+
+	// Arbitrary user-provided context for the event
+	Context string `json:"context,omitempty"`
+
+	// Messages of which type we should pass along; defaults to ["Alert"]
+	EventTypes []string `json:"eventTypes,omitempty"`
```

Comment:
Could validation be done to ensure valid types?

Reply:
Again, it would be more useful to validate here and provide feedback instead of relying on Ironic to fail the config.